packages/ui-text-area/src/TextArea/props.ts

```diff
@@ -39,110 +39,110 @@ import { TextareaHTMLAttributes } from 'react'
-  maxHeight?: number | string
-  messages?: FormMessage[]
-  inline?: boolean
-  placeholder?: string
-  disabled?: boolean
-  readOnly?: boolean
-  required?: boolean
-  textareaRef?: (...args: any[]) => any
-  defaultValue?: string
-  value?: any // TODO: controllable(PropTypes.string)
-  onChange?: (...args: any[]) => any
-}
-
-type PropKeys = keyof TextAreaOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type TextAreaProps = TextAreaOwnProps &
-  WithStyleProps<TextAreaTheme, TextAreaStyle> &
-  OtherHTMLAttributes<
-    TextAreaOwnProps,
-    TextareaHTMLAttributes<TextAreaOwnProps>
-  >
-
-type TextAreaStyle = ComponentStyle<
-  'textArea' | 'textAreaLayout' | 'textAreaOutline'
->
-
-const propTypes: PropValidators<PropKeys> = {
-  label: PropTypes.node.isRequired,
-  id: PropTypes.string,
   /**
    * sets the font-size for the textarea
    */
-  size: PropTypes.oneOf(['small', 'medium', 'large']),
-  layout: PropTypes.oneOf(['stacked', 'inline']),
+  size?: 'small' | 'medium' | 'large'
+  layout?: 'stacked' | 'inline'
   /**
    * the textarea will expand vertically to fit the height of the content,
    * unless its content exceeds `maxHeight`
    */
-  autoGrow: PropTypes.bool,
+  autoGrow?: boolean
   /**
    * is the textarea resizable (in supported browsers)
    */
-  resize: PropTypes.oneOf(['none', 'both', 'horizontal', 'vertical']),
+  resize?: 'none' | 'both' | 'horizontal' | 'vertical'
   /**
    * a fixed width for the textarea
    */
-  width: PropTypes.string,
+  width?: string
   /**
    * Initial height for the textarea (if autoGrow is true it will grow vertically)
    * Accepts CSS units, e.g. '55px'
    */
-  height: PropTypes.string,
+  height?: string
   /**
    * when autoGrow is true, the textarea will never grow beyond this value
    */
-  maxHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  maxHeight?: number | string
   /**
    * object with shape: `{
    * text: PropTypes.node,
    * type: PropTypes.oneOf(['error', 'hint', 'success', 'screenreader-only'])
    *   }`
    */
-  messages: PropTypes.arrayOf(FormPropTypes.message),
-  inline: PropTypes.bool,
+  messages?: FormMessage[]
+  inline?: boolean
   /**
    * Html placeholder text to display when the input has no value. This should be hint text, not a label
    * replacement.
    */
-  placeholder: PropTypes.string,
+  placeholder?: string
   /**
    * Whether or not to disable the textarea
    */
-  disabled: PropTypes.bool,
+  disabled?: boolean
   /**
    * Works just like disabled but keeps the same styles as if it were active
    */
-  readOnly: PropTypes.bool,
+  readOnly?: boolean
   /**
    * Sets the required property on the underlying native textArea
    */
-  required: PropTypes.bool,
+  required?: boolean
   /**
    * a function that provides a reference to the actual textarea element
    */
-  textareaRef: PropTypes.func,
+  textareaRef?: (textarea: HTMLTextAreaElement | null, ...args: any[]) => any
```

Comment:
The return here is `void`, and also the extra args might not be needed if you refactor the code in index.